url/src/lib.rs

```diff
@@ -1723,6 +1723,32 @@ impl Url {
                 );
             }
         });
+
+        // To handle cases like <non-spec:/> set pathname to </..//p>
+        // For instance, //p should be converted to /..//p here
+        // At this point, we would get "non-spec://p" for serialization
+        // and "/..//p" for path
+        if let Some(path_pos) = path.rfind("//") {
+            if let Some(serialization_pos) = self.serialization.rfind("//") {
+                const PATH_INCREMENT: usize = 2; // length of "/."
+
+                if path_pos + PATH_INCREMENT <= path.len()
+                    && serialization_pos + PATH_INCREMENT <= self.serialization.len()
```

Comment:
I'm not sure about these conditions.
Wouldn't they always be true?
Moreover,  I don't know if `path.rfind("//")` is the right check.
Step 3 of the serialization says:
> If url’s [host](https://url.spec.whatwg.org/#concept-url-host) is null, url does not have an [opaque path](https://url.spec.whatwg.org/#url-opaque-path), url’s [path](https://url.spec.whatwg.org/#concept-url-path)’s [size](https://infra.spec.whatwg.org/#list-size) is greater than 1, and url’s [path](https://url.spec.whatwg.org/#concept-url-path)[0] is the empty string, then append U+002F (/) followed by U+002E (.) to output.

Url's path size is greater that 1 and path[0] is empty string basically means path begins with `//`.
I don't think a rfind should be needed - if you think it is make sure to document it really well.

Reply:
This bug might need to be fixed after `has_host` is implemented, considering the case of:
```
"postgres://postgres@localhost"
"postgres:/.//postgres@localhost"
```